full-service/src/db/txo.rs

```diff
@@ -2056,13 +2058,10 @@ impl TxoModel for Txo {
             .left_join(transaction_output_txos::table)
             .left_join(txos::table.on(transaction_output_txos::txo_id.eq(txos::id)))
             .filter(transaction_output_txos::txo_id.eq(&self.id))
-            .filter(transaction_output_txos::is_change.eq(false))
             .filter(transaction_logs::failed.eq(false))
             .filter(transaction_logs::finalized_block_index.is_not_null())
             .filter(transaction_logs::submitted_block_index.is_not_null())
-            .filter(not(transaction_logs::account_id
-                .nullable()
-                .eq(txos::account_id)))
+            .filter(txos::account_id.is_null())
```

Comment:
this filter for secreted txos also suffered from the lack of an explicit check for txos.acount_id IS NULL.  We also decide for txo::status, which does not include the context of the account_id from which to consider the status of the txo, to return the status based on all that we know.  When txos.account_id is not null, then we have the account keys for this txo and we can differentiate between spent, unspent, unverified and orphaned.

Better to err on the side of the caller having the perspective of the txos owner rather than minter, and giving them the data they need.

Reply:
It might be worth commenting that this query results in secreted transaction logs specifically because the `account_id` field is the **recipient account id** (which I don't find intuitive and am likely to forget this specific definition if I see this code a month from now...). And given that its the recipient account, if it is null and the row indicates the txo was submitted+finalized+not failed, then it means it reached the blockchain and we don't know any more information about it (which is what Secreted stands for)